### DISCUSSION TITLE
Find a Z value, with a known X,Y

### DISCUSSION THREAD

**Discussion by MstrOfShadows** [pyvista/pyvista#1957] [Q&A] [answered]
Hello all, I have vtk polydata that contains x,y,z something akin to DEM/ elevation data. I was hoping to lookup my Z, is there a way to do this without implementing it myself? I all I've been able to find is FinClosestPoint() which gives the id of (x,y,x). Any info would be handy.

Thanks.

**Comment by RichardScottOZ** [comment]
You want to get the elevation at a particular xy point?

**Comment by RichardScottOZ** [comment, marked as answer]
Then if you have done the above, dem.points[id] to get your xyz

**Reply by AlexanderJuestel** [reply]
This seems interesting! I will definitely try that but what about Meshes that have multiple z values in one position, i.e. Geological layers like in GemPy? I will test it and report back :)

**Reply by MatthewFlamm** [reply]
Would this filter work?  https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSet.find_cells_along_line.html

**Reply by RichardScottOZ** [reply]
Assuming you have the line?

**Reply by MatthewFlamm** [reply]
If we know (x, y), then we can use the bounds to create the line:

```py
import pyvista
mesh = pyvista.UniformGrid(dims=(51, 51, 51), spacing = (1, 1, 1))
# cells that are located at (5.1, 5.1, *)
mesh.find_cells_along_line((5.1, 5.1, mesh.bounds[4]), (5.1, 5.1, mesh.bounds[5]))
```

**Reply by MatthewFlamm** [reply]
If you want points, you could extract the closest point in each intersecting cell for example.  But this gets tricky for 3D meshes.

**Comment by RichardScottOZ** [comment]
I was actually contemplating that yesterday when wondering about some polygons.
